src/_pytest/_code/code.py

```diff
@@ -609,9 +609,11 @@ def match(self, regexp: "Union[str, Pattern]") -> "Literal[True]":
         If it matches `True` is returned, otherwise an `AssertionError` is raised.
         """
         __tracebackhide__ = True
-        assert re.search(
-            regexp, str(self.value)
-        ), "Pattern {!r} does not match {!r}".format(regexp, str(self.value))
+        regextest = re.search(regexp, str(self.value))
+        msg = "Regex pattern {!r} does not match {!r}."
+        if regexp == str(self.value):
+            msg += " Did you mean to `re.escape()` the regex?"
+        assert regextest, msg.format(regexp, str(self.value))
```

Comment:
Take it or leave it: I always prefer to have the thing being checked directly in the assert, because then the exception is clearer. In this case it doesn't matter because it isn't visible to anyone directly, but just a habit.

```suggestion
        msg = "Regex pattern {!r} does not match {!r}."
        if regexp == str(self.value):
            msg += " Did you mean to `re.escape()` the regex?"
        assert re.search(regexp, str(self.value)), msg.format(regexp, str(self.value))
```